clumper/clump.py

```diff
@@ -950,6 +953,114 @@ def head(self, n=5):
         n = min(n, len(self))
         return self._create_new([self.blob[i] for i in range(n)])
 
+    def sample(
+        self,
+        n: int,
+        replace: bool,
+        random_state: Optional[int] = None,
+        weights: str = None,
+    ):
+        """Samples n data from the collection
+
+        Args:
+            n (int): The number of items to sample
+            replace (bool): Have duplicate items or not. Defaults to False.
+            weights (str, optional): The key(s) used for calculate sample probability. Defaults to None which means equal probability
+            random_state (int, optional): Random seed for reproducible results. Defaults to None.
+
+        Raises:
+            ValueError: Raises error when sampling more than size of collection
+
+        Returns:
+            Clumper: Sampled Clumper instance
+        """
+
+        # Sanity check:  n < len(clumper)
+        if n > len(self):
+            raise ValueError("n cannot be larger than the collection")
```

Comment:
Why would it not be possible if replace=True?

Reply:
I assumed that `random.choices` wouldn't take k bigger than input sequence. My assumption was incorrect. I will adjust my code accordingly.